src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/X509CertificateValidator.cs

```diff
@@ -54,6 +83,77 @@ public override void Validate(X509Certificate2 certificate)
             }
         }
 
+        class PeerTrustValidator : X509CertificateValidator
+        {
+            public override void Validate(X509Certificate2 certificate)
+            {
+                if (certificate == null)
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("certificate");
+
+                Exception exception;
+                if (!TryValidate(certificate, out exception))
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(exception);
+            }
+
+            static bool StoreContainsCertificate(StoreName storeName, X509Certificate2 certificate)
+            {
+                X509Store store = new X509Store(storeName, StoreLocation.CurrentUser);
+                X509Certificate2Collection certificates = null;
+                try
+                {
+                    store.Open(OpenFlags.ReadOnly);
+                    certificates = store.Certificates.Find(X509FindType.FindByThumbprint, certificate.Thumbprint, false);
+                    return certificates.Count > 0;
+                }
+                finally
+                {
+                    SecurityUtils.ResetAllCertificates(certificates);
+                    store.Dispose();
+                }
+            }
+
+            internal bool TryValidate(X509Certificate2 certificate, out Exception exception)
+            {
+                // Checklist
+                // 1) time validity of cert
+                // 2) in trusted people store
+                // 3) not in disallowed store
+
+                // The following code could be written as:
+                // DateTime now = DateTime.UtcNow;
+                // if (now > certificate.NotAfter.ToUniversalTime() || now < certificate.NotBefore.ToUniversalTime())
```

Comment:
I poked around the Unix PAL and Windows PAL and we are using local time. In fact, the Unix PAL specifically normalizes times to local (Windows returns it as local anyway). 

https://github.com/dotnet/corefx/blob/c2c725978f726e2058c61232d6d66628ab600366/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslCertificateFinder.cs#L91 shows that they're normalizing time for reporting. 

Creation of certificates should not be impacted here - if we see a DateTime.Kind != UTC, i believe we do the right thing and convert it to UTC. Ultimately, date/time stored in a certificate looks to be UTC anyway, so what we're doing is a "convenience" function, kind of - probably a relic of how Windows APIs reported time to begin with

Based on the code I'm inspecting - I think as long as we're consistent between how we're using time at each step, we should be okay. So I think we should just leave the code as is, even though Utc would probably be the better technological solution.